jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -35,291 +35,299 @@ YUI.add('juju-env-bakery', function(Y) {
-      },
-
-      /**
-       Takes the path supplied by the caller and makes a get request to the
-       requestHandlerWithInteraction instance. If setCookiePath is set then
-       it is used to set a cookie back to the ui after authentication.
-
-       @param {String} The path to make the api request to.
-       @param {Function} successCallback Called when the api request completes
-       successfully.
-       @param {Function} failureCallback Called when the api request fails
-       with a response of >= 400 except 401/407 where it does authentication.
-       @return {Object} The asynchronous request instance.
-       */
-      sendGetRequest: function (path, successCallback, failureCallback) {
-        var macaroons = this._getMacaroon();
-        var headers = {'Bakery-Protocol-Version': 1};
-        if (macaroons !== null) {
-          headers['Macaroons'] = macaroons;
-        }
-        return this.webhandler.sendGetRequest(
-          path, headers, null, null, false, null,
-          this._requestHandlerWithInteraction.bind(
-            this,
-            path,
-            successCallback,
-            failureCallback
-          )
-        );
-      },
-
-      /**
-       Takes the path supplied by the caller and makes a get request to the
-       requestHandlerWithInteraction instance. If setCookiePath is set then
-       it is used to set a cookie back to the ui after authentication.
-
-       @param path {String} The path to make the api request to.
-       @param data {String} Stringified JSON of parameters to send to the POST
-           endpoint.
-       @param successCallback {Function} Called when the api request completes
-           successfully.
-       @param failureCallback {Function} Called when the api request fails
-           with a response of >= 400 except 401/407 where it does
-           authentication.
-       @return {Object} The asynchronous request instance.
-       */
-      sendPostRequest: function (path, data, successCallback, failureCallback) {
-        var macaroons = this._getMacaroon();
-        var headers = {
-          'Bakery-Protocol-Version': 1,
-          'Content-type': 'application/json'
-        };
-        if (macaroons !== null) {
-          headers['Macaroons'] = macaroons;
-        }
-        return this.webhandler.sendPostRequest(
-          path, headers, data, null, null, false, null,
-          this._requestHandlerWithInteraction.bind(
-            this, path, successCallback, failureCallback)
+    Y.Base, [], {
+
+    /**
+     Initialize.
+
+     @method initializer
+     @param {Object} cfg A config object providing webhandler and visit method
+            information. A visitMethod can be provided, which becomes the
+            bakery's visitMethod. Alternatively, the bakery can be configured
+            to non interactive mode. If neither is true, the default method is
+            used.
+            {visitMethod: fn, interactive: boolean,
```

Comment:
This should document setCookiePath too, I think. And also what the contract for webhandler is.